src/libPMacc/include/particles/IdProvider.hpp

```diff
@@ -94,26 +106,33 @@ namespace PMacc {
     template<unsigned T_dim>
     bool IdProvider<T_dim>::isOverflown()
     {
-        // Get current value
-        uint64_t nextId = getNextId();
-        // Get start value
-        uint64_t globalUniqueStartId = getStartId();
-
-        /* Start value contains globally unique bits in the high order bits
-         * Hence compare all high order bits till the last set one and check,
-         * if any of them differs from the start value --> overflow
+        /* Overflow happens, when an id puts bits into the bits used for ensuring uniqueness.
+         * This are the n upper bits with n = highest bit set in the maximum id (which is maxNumProc_ - 1)
+         * when counting the bits from 1 = right most bit
+         * So first we calculate n, then remove the lowest bits of the next id so we have only the n upper bits
+         * If any of them is non-zero, it is an overflow and we can have duplicate ids.
+         * If not, then all ids are probably unique (still a chance, the id is overflown so much, that detection is impossible)
          */
-        // How far do we need to shift to get the highest bit
-        BOOST_STATIC_CONSTEXPR int bitsToHighest = sizeof(uint64_t) * CHAR_BIT - 1;
-        while(globalUniqueStartId)
+        uint64_t tmp = maxNumProc_ - 1;
+        int32_t bitsToCheck = 0;
+        while(tmp)
         {
-            // Compare highest bit
-            if(globalUniqueStartId >> bitsToHighest != nextId >> bitsToHighest)
-                return true;
-            // Proceed to next bit
-            globalUniqueStartId <<= 1;
+            bitsToCheck++;
+            tmp >>= 1;
         }
-        return false;
+
+        // Number of bits in the ids
+        BOOST_STATIC_CONSTEXPR int32_t numBitsOfType = sizeof(maxNumProc_) * CHAR_BIT;
+
+        // Get current value
+        uint64_t nextId = getState().get<0>();
+        /* Example: maxNumProc_ has 3 set bits (<8 ranks), 64bit value used
+         * --> Shift by 61 bits
+         * => 3 upper bits are left untouched (besides moving), rest is zero
+         */
+        nextId >>= numBitsOfType - bitsToCheck;
```

Comment:
offline discussed: 
- the current implementation is wrong
- we need the start id